python/cudf/cudf/core/frame.py

```diff
@@ -1449,9 +1449,12 @@ def sample(
                 columns, size=n, replace=replace, p=weights
             )
 
-            result = self[gather_map]
-            if not keep_index:
-                result.index = None
+            if isinstance(self, cudf.MultiIndex):
+                result = cudf.Index(self._source_data[gather_map])
```

Comment:
It's a shame we have to special case MultiIndex here. I guess when we refactor MultiIndex to inherit from Frame properly and have an associated `_data` attribute we could likely build a general solution instead.

Maybe add a `TODO` comment to address this when refactoring MultiIndex?

Reply:
I agree, will add a comment.